modules/database/sqlite/constants.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Define constants for the SQLite implementation.
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+// Temporary - This will be in wp-config.php once SQLite is merged in Core.
+if ( ! defined( 'DATABASE_TYPE' ) ) {
+	define( 'DATABASE_TYPE', 'sqlite' );
+}
+
+/**
+ * Notice:
+ * Your scripts have the permission to create directories or files on your server.
+ * If you write in your wp-config.php like below, we take these definitions.
+ * define('DB_DIR', '/full_path_to_the_database_directory/');
+ * define('DB_FILE', 'database_file_name');
+ */
+
+/**
+ * FQDBDIR is a directory where the sqlite database file is placed.
+ * If DB_DIR is defined, it is used as FQDBDIR.
+ */
+if ( ! defined( 'FQDBDIR' ) ) {
+	if ( defined( 'DB_DIR' ) ) {
+		define( 'FQDBDIR', trailingslashit( DB_DIR ) );
+	} elseif ( defined( 'WP_CONTENT_DIR' ) ) {
+		define( 'FQDBDIR', WP_CONTENT_DIR . '/database/' );
+	} else {
+		define( 'FQDBDIR', ABSPATH . 'wp-content/database/' );
+	}
+}
+
+/**
+ * FQDB is a database file name. If DB_FILE is defined, it is used
+ * as FQDB.
+ */
+if ( ! defined( 'FQDB' ) ) {
+	if ( defined( 'DB_FILE' ) ) {
+		define( 'FQDB', FQDBDIR . DB_FILE );
+	} else {
+		define( 'FQDB', FQDBDIR . '.ht.sqlite' );
```

Comment:
You chose a hidden (.) file for the .sqlite database container. Is that intentional? Does it make sense to make it hard to find?  Also, is the subdirectory important?

Reply:
That part was copied verbatim from https://github.com/aaemnnosttv/wp-sqlite-db/blob/f94ae45ed1ffa28ec5e64c6f137c557b2446069b/src/db.php#L123, but I believe it makes sense... Just an additional security measure